### DISCUSSION TITLE
File per Document and Folder per Table

### DISCUSSION THREAD

**Discussion by JonLangfordUK** [msiemens/tinydb#503] [Q&A]
We're looking at using TinyDB for use as the backend for a reference system, but we need this solution to be compatible via our source control solution.

Ideally we'd have a file per record as well as a folder per table so that we can more easily manage checkout with source control. Is this possible with the current version of TinyDB? If not, is this something that we could easily modify/add ourselves.

**Comment by msiemens** [comment]
Hey @JonLangfordUK, I think implementing a scheme like this should be possible by writing a [custom storage class](https://tinydb.readthedocs.io/en/latest/extend.html#write-a-custom-storage). Regarding source control: you'll need to keep in mind that TinyDB doesn't particularly like other systems modifying its data behind its back. Doing so might result in data loss or corruption, e.g. if you update the data while TinyDB tries to update it.
